app/routes/formbuilder/edit.js

```diff
@@ -12,17 +12,41 @@ export default class FormbuilderEditRoute extends Route {
     this.registerCustomFields();
   }
 
-  model(params) {
-    return this.store.findRecord('generated-form', params.id);
+  async model(params) {
+    return await this.getGeneratedFormById(params.id);
   }
 
   setupController(controller, model) {
     super.setupController(...arguments);
     controller.refresh.perform({
-      value: model.ttlCode ? model.ttlCode : template,
+      formTtlCode: this.getFormTtlCode(model),
+      resetBuilder: false,
+      isInitialRouteCall: true,
     });
   }
 
+  resetController(controller) {
+    if (
+      controller.builderStore &&
+      controller.builderStore.observers[controller.REGISTERED_FORM_TTL_CODE_KEY]
+    ) {
+      controller.builderStore.deregisterObserver();
```

Comment:
This still needs the `key` to properly remove the observer.

What I like to do is add a `reset` method to the controller and call that here. That way the controller can reset what it needs without the route having to know all that. That would also allow you to easily access the key constant you defined.

Reply:
A good way of thinking thanks! [Changed](https://github.com/lblod/frontend-form-builder/pull/14/commits/955389ac216abe6c0090abc4d57fec37d5064658) it so the router just calls the controller method :ok_hand: